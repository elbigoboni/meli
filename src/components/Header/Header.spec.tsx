import React from "react";
import { render, screen } from "@testing-library/react";
import { Header } from ".";
import { MemoryRouter } from "react-router-dom";

describe("<Header />", () => {
  beforeEach(() => {
    render(<Header />, { wrapper: MemoryRouter });
  });
  describe("Render", () => {
    test("should render correctly", () => {
      const { container } = render(<Header />, { wrapper: MemoryRouter });
      expect(container).toMatchSnapshot();
    });
    test("should render link to home", () => {
      const linkElement = screen.getByTestId("link to home");
      expect(linkElement).toBeInTheDocument();
    });
    test("should render link to home with path to '/'", () => {
      const linkElement = screen.getByTestId("link to home");
      expect(linkElement).toHaveAttribute("href", "/");
    });
    test("should render <SearchBox />", () => {
      const searchBox = screen.getByTestId("search box");
      expect(searchBox).toBeInTheDocument();
    });
  });
});
