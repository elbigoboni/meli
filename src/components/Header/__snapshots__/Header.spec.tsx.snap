// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Render should render correctly 1`] = `
<div>
  <header
    class="sc-jSgupP gRFoCY"
  >
    <header
      class="sc-eCssSg dznMVN"
    >
      <div
        class="sc-iBPRYJ iUImDk"
      >
        <a
          data-testid="link to home"
          href="/"
        >
          <img
            class="sc-gKsewC fVMCuu"
            data-testid="logo"
            src="https://http2.mlstatic.com/frontend-assets/ui-navigation/5.10.4/mercadolibre/logo__small@2x.png"
          />
        </a>
        <form
          class="sc-bdfBwQ ggUMEn"
          data-testid="search box"
        >
          <div
            class="sc-hKgILt gFyfpC"
          >
            <input
              class="sc-gsTCUz iPwkQd"
              data-testid="search field"
              placeholder="Nunca dejes de buscar"
              type="text"
              value=""
            />
            <button
              class="sc-dlfnbm daFmpW"
            >
              GO
            </button>
          </div>
        </form>
      </div>
    </header>
  </header>
</div>
`;
